sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/ChunkFactory.java

```diff
@@ -0,0 +1,94 @@
+package com.azure.storage.blob.changefeed;
+
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.changefeed.implementation.models.ChangefeedCursor;
+import com.azure.storage.common.implementation.Constants;
+import com.azure.storage.common.implementation.StorageImplUtils;
+import com.azure.storage.internal.avro.implementation.AvroReader;
+import com.azure.storage.internal.avro.implementation.AvroReaderFactory;
+import reactor.core.publisher.Flux;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Factory class for {@link Chunk}.
+ */
+class ChunkFactory {
+
+    /* TODO (gapra): This should probably be configurable by a user. */
```

Comment:
By that comment I was referring to the chunk size (1MB). I placed it poorly. The performance of the downloads can vary so in case a customer is like - this is super slow, we can allow them to increase that threshold or decrease to improve perf depending on their situation. I think it would be a nice to have kind of thing